src/ucs/vfs/fuse/vfs_fuse.c

```diff
@@ -164,13 +164,36 @@ static int ucs_vfs_fuse_release(const char *path, struct fuse_file_info *fi)
     return 0;
 }
 
+static int ucs_vfs_fuse_write(const char *path, const char *buf, size_t size,
+                              off_t offset, struct fuse_file_info *fi)
+{
+    ucs_status_t status;
+
+    if (offset > 0) {
+        ucs_warn("cannot write to %s with non-zero offset", path);
+        return 0;
+    }
+
+    status = ucs_vfs_path_write_file(path, buf, size);
+
+    if (status == UCS_ERR_NO_ELEM) {
+        return -ENOENT;
+    } else if (status == UCS_ERR_INVALID_PARAM) {
+        return -EINVAL;
+    }
+    ucs_assert(status == UCS_OK);
```

Comment:
if status can be anything (as returned from user cb), can add `} else { return -EIO }`